### PR TITLE
Add configurable max heap setting (Fixes #1624)

### LINKED ISSUE #1624: Add configurable max heap size setting to settings UI and settings.json
## Context

In #1622 we capped the auto-configured --max-old-space-size at 8GB (previously it was uncapped at 50% of system RAM, which meant 64GB on a 128GB machine). The 8GB cap is a sensible default but power users with very large conversations or heavy tool output may want to raise or lower it.

## Current State

- autoConfigureMaxOldSpaceSize (boolean) exists in settings schema under ui -- it toggles the feature on/off but does not allow setting a specific value
- The cap is hardcoded as MAX_HEAP_CAP_MB = 8192 in bootstrap.ts
- No way for users to override the cap without disabling auto-configuration entirely

## Requested Enhancement

Add a new numeric setting (e.g. maxHeapSizeMB) that:

1. Appears in the /settings menu and settings.json under the UI category
2. Defaults to 8192 (8GB) matching the current cap
3. Is read by shouldRelaunchForMemory() and computeSandboxMemoryArgs() in bootstrap.ts instead of the hardcoded MAX_HEAP_CAP_MB
4. Requires restart (like autoConfigureMaxOldSpaceSize)
5. Has reasonable validation (minimum 512MB, no maximum -- let users shoot themselves in the foot if they want)

## Challenge

bootstrap.ts runs BEFORE settings are loaded (that is the whole point -- it decides whether to relaunch before doing heavy init). The setting would need to be read from the settings.json file directly during bootstrap, similar to how autoConfigureMaxOldSpaceSize is checked in gemini.tsx after settings load but before relaunch. This may require restructuring the bootstrap flow slightly.

## Related

- #1622 (introduced the 8GB cap)

### PR DESCRIPTION
**Pull request by acoliver**
## TLDR

Adds a UI/settings.json setting, ui.maxHeapSizeMB, so users can configure the cap used by automatic Node.js max old space sizing instead of being limited to the previous hardcoded 8192 MB cap.

## Dive Deeper

This change:

- Adds ui.maxHeapSizeMB to the UI settings schema with default 8192, minimum 512, integer-only validation, restart requirement, and dialog visibility.
- Passes the configured value into both non-sandbox relaunch memory sizing and sandbox process memory argument computation.
- Keeps MAX_HEAP_CAP_MB as the fallback/default cap for direct utility callers.
- Extends schema validation and generated settings schema support for numeric multipleOf constraints.
- Adds defensive runtime flooring so fractional direct utility inputs cannot emit fractional max old space arguments.
- Validates numeric edits in the settings dialog against schema minimum, maximum, and multipleOf constraints before accepting pending values.
- Adds tests for schema behavior, settings utility behavior, CLI wiring, sandbox wiring, bootstrap cap calculation, type inference, and validation edge cases.

## Reviewer Test Plan

Reviewers can validate by:

1. Running the verification commands listed below.
2. Opening the settings UI and confirming Max Heap Size (MB) appears under UI settings.
3. Setting ui.maxHeapSizeMB to a valid integer such as 4096 and restarting; automatic max old space sizing should use that cap.
4. Trying invalid values such as 511 or 1536.75; validation should reject them.
5. Using sandbox mode with auto memory configuration enabled and confirming the sandbox process receives memory args computed with the configured cap.

## Testing Matrix

|          |   |   |   |
| -------- | --- | --- | --- |
| npm run  | [OK]  |   |   |
| npx      |   |   |   |
| Docker   |   |   |   |
| Podman   |   | -   | -   |
| Seatbelt |   | -   | -   |

Verified on macOS:

- npm run test
- npm run lint
- npm run typecheck
- npm run format
- npm run build
- node scripts/start.js --profile-load ollamakimi "write me a haiku and nothing else"

## Linked issues / bugs

Fixes #1624

### CODE REVIEW

File: packages/cli/src/utils/settingsUtils.test.ts [hunk 1]
```diff
@@ -71,12 +71,49 @@ describe('SettingsUtils', () => {
...
+        expect(Number.isInteger(value / definition!.multipleOf!)).toBe(false);
+      });
+
+      it('should accept value exactly at minimum', () => {
+        const definition = getSettingDefinition('ui.maxHeapSizeMB');
+        const value = 512;
+        expect(value).toBeGreaterThanOrEqual(definition!.minimum!);
+        expect(Number.isInteger(value / definition!.multipleOf!)).toBe(true);
+      });
     });
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/utils/settingsUtils.test.ts]: Addressed. I removed the redundant settingsUtils numeric validation assertions and kept that file focused on settings utility contract coverage. The actual validation behavior remains covered through validateSettings tests in settingsSchema.test.ts for accepted integer values, minimum boundary, below-minimum rejection, and fractional rejection. Full local verification passed after the change: test, lint, typecheck, format, build, and the ollamakimi smoke test.
